person.rb

```diff
@@ -13,9 +16,38 @@ def can_use_services?
     of_age? || @parent_permission
   end
 
+  def correct_name
+    name
+  end
+
   private
 
   def of_age?
     @age >= 18
   end
 end
+
+class Decorator < Nameable
+  attr_accessor :component
+
+  def initialize(component)
+    super
+    @component = component
+  end
+
+  def correct_name
+    @component.correct_name
+  end
+end
+
+class CapitalizeDecorator < Decorator
+  def correct_name
+    @component.correct_name.capitalize
+  end
+end
+
+class TrimmerDecorator < Decorator
+  def correct_name
+    @component.correct_name.capitalize[0..9]
```

Comment:
- [ ] This `TrimmerDecorator` should only trim the name if exceeded 10 characters, not also capitalize the name because there is another decorator for that. So please fix that.

[Hint]: The best (sophisticated) way to do it:
```ruby
def correct_name
      @nameable.correct_name.length > 10 ? @nameable.correct_name[0..9] : @nameable.correct_name
end
```
or you can simply do it like that:
```ruby
def correct_name
     @component.correct_name[0..9]
end
```

Reply:
You are right. An oversight on my part. 

Also thanks for your hint. That's def a better way to do it.